client/src/app/components/stock-historical-chart/stock-historical-chart.component.ts

```diff
@@ -0,0 +1,224 @@
+        left: '2%',
+        right: '2%',
+        bottom: '2%',
+        top: '5%',
+        containLabel: true,
+      },
+      xAxis: [
+        {
+          type: 'category',
+          boundaryGap: false,
+          data: this.xAxisData,
+        },
+      ],
+      yAxis: [
+        {
+          type: 'value',
+          interval: this.intervalValue,
+          min: this.minValue,
+          max: this.maxValue,
+        },
+      ],
+      series: [
+        {
+          name: 'Price (' + this.historicalData.currency + ')',
+          type: 'line',
+          stack: 'u',
+          showSymbol: false,
+          areaStyle: {
+            opacity: 0.8,
+            color: new graphic.LinearGradient(0, 0, 0, 1, [
+              {
+                offset: 0,
+                color: 'rgba(7, 54, 124, 100)',
+              },
+              {
+                offset: 1,
+                color: 'rgba(1, 191, 236, 0)',
+              },
+            ]),
+          },
+          emphasis: {
+            focus: 'series',
+          },
+          data: this.yAxisData,
+          connectNulls: true,
+        },
+      ],
+    };
+  }
+
+  async formatChartData(): Promise<void> {
+    this.xAxisData = this.convertTimeStampToDate(this.historicalData.timestamp);
+    this.yAxisData = this.formatValues(this.historicalData.closePrice);
+  }
+
+  convertTimeStampToDate(time: any[]): string[] {
+    const xAxis: string[] = [];
+    let conversionTime: Date;
+    const months = [
+      'Jan',
+      'Feb',
+      'Mar',
+      'Apr',
+      'May',
+      'Jun',
+      'Jul',
+      'Aug',
+      'Sep',
+      'Oct',
+      'Nov',
+      'Dec',
+    ];
+
+    time.forEach((t) => {
+      conversionTime = new Date(t * 1000);
+      const year = conversionTime.getFullYear();
+      const month = months[conversionTime.getMonth()];
+      const date = conversionTime.getDate();
+      const hour = String(conversionTime.getHours());
+      const min = String(conversionTime.getMinutes());
+
+      switch (this.selectedRange) {
+        case '1d':
+        case '5d':
+        case '1mo':
+          xAxis.push(
+            String(
+              month +
+                ' ' +
+                date +
+                '\n ' +
+                hour.padStart(2, '0') +
+                ':' +
+                min.padStart(2, '0')
+            )
+          );
+          break;
+        case '6mo':
+        case '1y':
+        case '5y':
+        case 'max':
+          xAxis.push(String(month + ' ' + date + '\n ' + year));
+          break;
+      }
+    });
+
+    return xAxis;
+  }
+
+  changeRangeInterval(option: ChartViewOptions): void {
+    this.selectedRange = option.range;
+    this.changeHistoricalChartRangeInterval.emit({
+      range: option.range,
+      interval: option.interval,
+    });
+  }
+
+  getOptionClass(option: string): string {
+    if (this.selectedRange === option) {
+      return 'range-option-selected';
+    }
+    return 'range-option';
+  }
```

Comment:
This can be changed to:
return this.selectRange === option ? 'range-option-selected' : 'range-option';